NMS-13892: BMP Introduction (#6596)
OpenNMS/opennms: docs/modules/reference/pages/telemetryd/protocols/bmp/bmp.adoc

The https://datatracker.ietf.org/doc/html/rfc7854[BGP Monitoring Protocol] (BMP) monitors Border Gateway Protocol (BGP) sessions and provides a convenient interface to monitor BGP routing information on the routing device.

The BMP integration in {page-component-title} lets you use these route views, status updates, and statistics for advanced monitoring and management, providing visibility into what BGP is doing.

== {page-component-title} BMP integration

{page-component-title} leverages https://www.snas.io/[OpenBMP], an application that was developed as a reference implementation for receiving BMP messages from network equipment and presenting the state of the BGP network to the user.

BGP runs on routers and high-end switches.
Those devices can send BMP data to {page-component-title} directly or through a {page-component-title} Minion collector.

.BMP integration overview
image::bmp/bmp_integration_overview_2.png[width=70%]

The {page-component-title} BMP parser accepts BMP connections from router packets using a TCP listener.

> Three adapters in OpenNMS read BMP data, including an inventory list of connected routers and the peers they are established with, and stats and data about what is being exchanged over those peers:

Reviewer: Since we longer referencing OpenBMP Integration Adapter, Three adapaters might be wrong.

Author: Doh! Yes. Thanks for the catch.

= BGP Monitoring Protocol
:description: Learn how the BGP Monitoring Protocol (BMP) integration in OpenNMS Horizon/Meridian helps you to see what the Border Gateway Protocol is doing.

The https://datatracker.ietf.org/doc/html/rfc7854[BGP Monitoring Protocol] (BMP) monitors Border Gateway Protocol (BGP) sessions and provides a convenient interface to monitor BGP routing information on the routing device.

The BMP integration in {page-component-title} lets you use these route views, status updates, and statistics for advanced monitoring and management, providing visibility into what BGP is doing.

> == OpenNMS BMP integration

Reviewer 2: ```suggestion
== {page-component-title} BMP integration
```

> OpenNMS leverages https://www.snas.io/[OpenBMP], an application that was developed as a reference implementation for receiving BMP messages from network equipment and presenting the state of the BGP network to the user.

Reviewer 2: ```suggestion
{page-component-title} leverages https://www.snas.io/[OpenBMP], an application that was developed as a reference implementation for receiving BMP messages from network equipment and presenting the state of the BGP network to the user.
```

BGP runs on routers and high-end switches.

> Those devices can send BMP data to {page-component-title} directly or through an OpenNMS Minion collector.

Reviewer 2: ```suggestion
Those devices can send BMP data to {page-component-title} directly or through a {page-component-title} Minion collector.
```

> Those devices can point to {page-component-title} or an OpenNMS Minion and send BMP data to {page-component-title}.

Reviewer 3: "point to opennms horizon or an opennms minion and send bmp data to opennms horizon" feels a little awkward? not sure how to change it tho

Author: Good catch. Thanks. I'll see what I can do. No worries about the placeholder questions. Perhaps Mark knows.

.BMP integration overview
image::bmp/bmp_integration_overview_2.png[width=70%]

> The OpenNMS BMP parser accepts BMP connections from router packets using a TCP listener.

Reviewer 2: ```suggestion
The {page-component-title} BMP parser accepts BMP connections from router packets using a TCP listener.
```

> The OpenNMS BMP parser accepts BMP connections from router packets using a TCP listener.
> Three adapters in OpenNMS read BMP data, including an inventory list of connected routers and the peers they are established with, and stats and data about what is being exchanged over those peers:

Reviewer 2: ```suggestion
The {page-component-title} BMP parser accepts BMP connections from router packets using a TCP listener.
Three adapters read BMP data, including an inventory list of connected routers and the peers they are established with, and stats and data about what is being exchanged over those peers:
```

Author: Had to move this change to the other comment, as it wouldn't let me commit multiple changes to the same line.

> Three adapters in OpenNMS read BMP data, including an inventory list of connected routers and the peers they are established with, and stats and data about what is being exchanged over those peers:

Reviewer 2: ```suggestion
Three adapters in {page-component-title} read BMP data, including an inventory list of connected routers and the peers they are established with, and stats and data about what is being exchanged over those peers:
```

Author: Edited previous conversation to include this change, as could not merge with two changes to the same line.

* *BMP Telemetry Adapter:* Handles BMP statistics the BMP parser has received and parsed.
Statistics received from the router are associated as performance data with that router.
View metrics and statistics related to that peer and apply thresholding.

* *BMP Peer Status Adapter:* Handles BMP Peer Up and Down messages that the BMP parser receives and parses, and converts them to OpenNMS events.
Allows alarm creation based on up/down status.

//I notice that Integration Adapter is missing from the diagram. Should we add it? Has the architecture changed?

> * *OpenBMP Integration Adapter:* Integrates with an existing OpenBMP installation and handles BMP messages from the BMP parser, and creates OpenBMP-compatible messages, which are then passed to the OpenBMP Kafka cluster.

Reviewer: OpenBMP Integration Adapter may not be relevant any more.

The parser and adapters are configurable in the telemetryd-configuration.xml file.

== Before you begin

> To use OpenNMS BMP features, you need the OpenBMP https://github.com/SNAS/obmp-postgres[PostgreSQL component] installed and set up, and have a BMP-enabled router.

Reviewer 2: ```suggestion
To use {page-component-title} BMP features, you need the OpenBMP https://github.com/SNAS/obmp-postgres[PostgreSQL component] installed and set up, and have a BMP-enabled router.
```

> To use {page-component-title} BMP features, you need the OpenBMP https://github.com/SNAS/obmp-postgres[PostgreSQL component] installed and set up, and have a BMP-enabled router.

Reviewer: we don't need OpenBMP postgres component. I think this is only needed for OpenBMP Integration Adapter.
